site/src/theme/mui.ts

```diff
@@ -1,3 +1,18 @@
+/* eslint-disable @typescript-eslint/no-explicit-any
+-- we need to hack around the MUI types a little */
```

Comment:
It looks like we can get a little more specific with the type assertions by doing something like this:
```ts
type MuiOverrideStyleKey = keyof ComponentNameToClassKey;

styleOverrides: {
  ["sizeXlarge" as MuiOverrideStyleKey]: {...}
}
```
I pulled down the branch and tried it out, and it *seems* like it silences the compiler warnings. You know more about the theming set up than I do, though. I'm normally pretty skittish around `any` types, but do you think a change like this is even worth it?

Reply:
my concern about that is that it's technically lying to the compiler 😅 we'd be casting it as something it is not, and it would read as "trust me, I know what I'm doing", whereas using `any` is hinting a bit more at "this _should_ make you nervous because it's wrong and we should maybe fix it more rigorously later"

Reply:
(I genuinely intend to improve this in the new year. I hope I've earned a bit of trust that I'm the type of dev who will actually go back and clean up debt, lol.)

Reply:
I guess the way I see it is that both ways are lying to the compiler, but the only the `any` approach completely turns the compiler off

I'm cool if you want to keep things as-is, but if anything, I feel like the way to split the difference would be to rename the type to something like `HackyMuiOverrideStyleKey`

Reply:
Oh, sorry. For some reason your second comment didn't come in
To be clear, I do trust you – I didn't realize that this was a temp solution